src/sql/parts/notebook/outputs/renderers.ts

```diff
@@ -0,0 +1,630 @@
+
+      // Copy the text content into the clone.
+      clone.textContent = script.textContent;
+
+      // Replace the old script in the parent.
+      script.parentNode.replaceChild(clone, script);
+    }
+  }
+
+  /**
+   * Handle the default behavior of nodes.
+   */
+  export function handleDefaults(
+    node: HTMLElement,
+    resolver?: IRenderMime.IResolver
+  ): void {
+    // Handle anchor elements.
+    let anchors = node.getElementsByTagName('a');
+    for (let i = 0; i < anchors.length; i++) {
+      let path = anchors[i].href || '';
+      const isLocal =
+        resolver && resolver.isLocal
+          ? resolver.isLocal(path)
+          : URLExt.isLocal(path);
+      if (isLocal) {
+        anchors[i].target = '_self';
+      } else {
+        anchors[i].target = '_blank';
+      }
+    }
+
+    // Handle image elements.
+    let imgs = node.getElementsByTagName('img');
+    for (let i = 0; i < imgs.length; i++) {
+      if (!imgs[i].alt) {
+        imgs[i].alt = 'Image';
+      }
+    }
+  }
+
+  /**
+   * Resolve the relative urls in element `src` and `href` attributes.
+   *
+   * @param node - The head html element.
+   *
+   * @param resolver - A url resolver.
+   *
+   * @param linkHandler - An optional link handler for nodes.
+   *
+   * @returns a promise fulfilled when the relative urls have been resolved.
+   */
+  export function handleUrls(
+    node: HTMLElement,
+    resolver: IRenderMime.IResolver,
+    linkHandler: IRenderMime.ILinkHandler | null
+  ): Promise<void> {
+    // Set up an array to collect promises.
+    let promises: Promise<void>[] = [];
+
+    // Handle HTML Elements with src attributes.
+    let nodes = node.querySelectorAll('*[src]');
+    for (let i = 0; i < nodes.length; i++) {
+      promises.push(handleAttr(nodes[i] as HTMLElement, 'src', resolver));
+    }
+
+    // Handle anchor elements.
+    let anchors = node.getElementsByTagName('a');
+    for (let i = 0; i < anchors.length; i++) {
+      promises.push(handleAnchor(anchors[i], resolver, linkHandler));
+    }
+
+    // Handle link elements.
+    let links = node.getElementsByTagName('link');
+    for (let i = 0; i < links.length; i++) {
+      promises.push(handleAttr(links[i], 'href', resolver));
+    }
+
+    // Wait on all promises.
+    return Promise.all(promises).then(() => undefined);
+  }
+
+  /**
+   * Apply ids to headers.
+   */
+  export function headerAnchors(node: HTMLElement): void {
+    let headerNames = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
+    for (let headerType of headerNames) {
+      let headers = node.getElementsByTagName(headerType);
+      for (let i = 0; i < headers.length; i++) {
+        let header = headers[i];
+        header.id = encodeURIComponent(header.innerHTML.replace(/ /g, '-'));
+        let anchor = document.createElement('a');
+        anchor.target = '_self';
+        anchor.textContent = '¶';
+        anchor.href = '#' + header.id;
+        anchor.classList.add('jp-InternalAnchorLink');
```

Comment:
Please create tracking tasks to support these CSS types

Reply:
Task #3156